Mathlib/Probability/Kernel/Disintegration/Unique.lean

```diff
@@ -0,0 +1,178 @@
+/-
+Copyright (c) 2023 Kexing Ying. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kexing Ying, Rémy Degenne
+-/
+import Mathlib.Probability.Kernel.Disintegration.Integral
+
+/-!
+# Uniqueness of the conditional kernel
+
+We prove that the conditional kernels `ProbabilityTheory.kernel.condKernel` and
+`MeasureTheory.Measure.condKernel` are almost everywhere unique.
+
+## Main statements
+
+* `ProbabilityTheory.eq_condKernel_of_kernel_eq_compProd`: a.e. uniqueness of
+  `ProbabilityTheory.kernel.condKernel`
+* `ProbabilityTheory.eq_condKernel_of_measure_eq_compProd`: a.e. uniqueness of
+  `MeasureTheory.Measure.condKernel`
+* `ProbabilityTheory.kernel.condKernel_apply_eq_condKernel`: the kernel `condKernel` is almost
+  everywhere equal to the measure `condKernel`.
+-/
+
+open MeasureTheory Set Filter MeasurableSpace
+
+open scoped ENNReal MeasureTheory Topology ProbabilityTheory
+
+namespace ProbabilityTheory
+
+variable {α β Ω : Type*} {mα : MeasurableSpace α} {mβ : MeasurableSpace β}
+  [MeasurableSpace Ω] [StandardBorelSpace Ω] [Nonempty Ω]
+
+section Measure
+
+variable {ρ : Measure (α × Ω)} [IsFiniteMeasure ρ]
+
+/-! ### Uniqueness of `Measure.condKernel`
+
+The conditional kernel of a measure is unique almost everywhere. -/
+
+/-- A s-finite kernel which satisfy the disintegration property of the given measure `ρ` is almost
+everywhere equal to the disintegration kernel of `ρ` when evaluated on a measurable set.
+
+This theorem in the case of finite kernels is weaker than `eq_condKernel_of_measure_eq_compProd`
+which asserts that the kernels are equal almost everywhere and not just on a given measurable
+set. -/
+theorem eq_condKernel_of_measure_eq_compProd' (κ : kernel α Ω) [IsSFiniteKernel κ]
```

Comment:
I think I am unhappy with the assumptions in this theorem and in the next ones: if `ρ = ρ.fst ⊗ₘ κ` and `ρ` is a finite measure, then `κ a` should have finite mass almost everywhere wrt `ρ.fst`, and I feel that this should be sufficient to carry out the arguments in the lemma without the assumptions of s-finiteness or of finiteness. However, the lemmas were already there (at least in the measure version) and you just copied them, so this shouldn't hold the PR back. I'll investigate that once this one is merged, if you agree.